sql/parser/eval_test.go

```diff
@@ -250,8 +250,8 @@ func TestEvalExpr(t *testing.T) {
 		{`'2010-09-28'::timestamp`, `2010-09-28 00:00:00+00:00`},
 		{`('2010-09-28 12:00:00.1'::timestamp)::date`, `2010-09-28`},
 		{`'2010-09-28 12:00:00.1'::timestamp`, `2010-09-28 12:00:00.1+00:00`},
-		{`'2010-09-28 12:00:00.1+02:00'::timestamp`, `2010-09-28 10:00:00.1+00:00`},
-		{`'2010-09-28 12:00:00.1-07:00'::timestamp`, `2010-09-28 19:00:00.1+00:00`},
+		{`'2010-09-28 12:00:00.1+02:00'::timestamp`, `2010-09-28 12:00:00.1+02:00`},
```

Comment:
We want to use the principle--time zone is only used for parsing and presentation. The presentation is happening on the client side and is a separate issue. The parsing is happening on the server side, and ideally is the only place where we need to change time/date to UTC. 

why does decodeTableKey need to know the timezone? The encoded key for a date is a time at      0 hours UTC for that date.